tests/streamlit_ui_tests/features/merge.spec.ts

```diff
@@ -79,4 +79,144 @@ test.describe('Merge', () => {
         // Check that there are 5 columns still
         await expect(mito.locator('.endo-column-header-container')).toHaveCount(5);
     });
+
+    test('Change Merge Type', async ({ page }) => {
+        const mito = await getMitoFrame(page);
+        await importCSV(page, mito, 'merge.csv');
+        await importCSV(page, mito, 'test.csv');
+        
+        await clickButtonAndAwaitResponse(page, mito, { name: '▾ Merge' })
+        await mito.getByText('Merge (horizontal)').click();
+        await awaitResponse(page);
+    
+        await expect(mito.getByText('Merge Dataframes')).toBeVisible();
+
+        // Wait for the merge to be finished before continuing so adding a column works!
+        await expect(mito.getByText('df_merge')).toBeVisible();
+    
+        // Check that the correct number of rows are present
+        await hasExpectedNumberOfRows(mito, 4);
+
+        const changeMergeType = async (oldMergeType: string, newMergeType: string, expectedRows: number) => {
+            // Change the merge type
+            await mito.locator('.select-text', { hasText: oldMergeType }).click();
+            await mito.locator('.mito-dropdown-item span').getByText(newMergeType, { exact: true }).click();
+            await awaitResponse(page);
+
+            // Check that the correct number of rows are present
+            await hasExpectedNumberOfRows(mito, expectedRows);
+        }
```

Comment:
Great utility. Can we refactor this to:
1. Just take the newMergeType (so you'll have to change how you're locating things)
2. Not do expected rows
3. Make this a global utility in the file


You can then make this little local one for checking rows as well, if you wanna keep the lines of code below the same -- but the global utility seems so good!